lib/procore-sift.rb

```diff
@@ -39,20 +39,27 @@ def filter_errors
   def filter_validator
     @_filter_validator ||= FilterValidator.build(
       filters: filters,
-      sort_fields: self.class.sort_fields,
+      sort_fields: sort_fields,
       filter_params: filter_params,
       sort_params: sort_params,
     )
   end
 
   def filters
-    self.class.filters
+    self.class.ancestors
+      .flat_map { |klass| klass.try(:filters) }
+      .compact
+      .uniq { |f| [f.param, f.class] }
```

Comment:
Given that a controller may have a large number of ancestors depending on the code base (our Vapid Incidents controller has 101), it might be worth trying to scope this down to some degree. However, I'm not sure if there's a way to avoid overhead here.

Reply:
I scoped down the lookup by dropping everything after the class named`Sift`, which should exclude the boilerplate classes. I don't know what/if there is a performance gain, but it couldn't hurt. I'll make sure all the tests still pass in Procore with this.